### ISSUE TITLE
模型效果复现相关

### ISSUE THREAD

**Issue by Hans2573** [YangZhengyi98/DreamRec#6]
对代码中提供的zhihu数据集，进行相关实验：

使用README.md中给出的命令：
```python -u DreamRec.py --data zhihu --timesteps 500 --lr 0.01 --beta_sche linear --w 4 --optimizer adamw --diffuser_type mlp1 --random_seed 100 ```

达不到论文中的效果，HR@20 < 2.00%，NDCG@20<0.70%，

使用optuna调参后，得到最大的ndcg@20效果；
![image](https://github.com/YangZhengyi98/DreamRec/assets/70685549/a8a7d45d-5126-4a4b-9896-73f8574c4b44)

ndcg@20次优结果：
![image](https://github.com/YangZhengyi98/DreamRec/assets/70685549/2b6d016f-f86d-40ac-94cc-426293273a23)

不过所有的HR@20 均达不到论文中给出的效果，（调参实验中：HR@20 <= 0.020380）

调参范围严格按照论文中的：
“We leverage AdamW as the optimizer. The embedding dimension of items is fixed as 64 across all models. The learning rate is tuned in the range of [0.01, 0.005, 0.001, 0.0005, 0.0001, 0.00005]. Despite that DreamRec does not require L2 regularization, we tune the weight of L2 regularization for all baselines in the range of [1e-3, 1e-4, 1e-5, 1e-6, 1e-7]. For all baselines, we conduct negative sampling from the uniform distribution at the ratio of 1: 1, which is not conducted in DreamRec. For our DreamRec, we fix the unconditional training probability pu as 0.1 suggested by [36]. We search the total diffusion step T in the range of [50, 100, 200, 500, 1000, 2000], and the personalized guidance strength w in the range of [0, 2, 4, 6, 8, 10].”

论文中的模型效果参数是如何设置？感谢。

**Comment by Yewx0310** [comment]
+1，我跑zhihu也跑不出来论文中的结果

**Comment by YangZhengyi98** [comment]
你好，

可能是由于实验环境不同，在论文中给定的实验环境下(GPU type: 3090, python==3.7.13, pytorch==1.12.1, numpy==1.21.5)，按提供参数的实验结果为：

![8660cd9c52f8d2dd1aaf4950c6131ae](https://github.com/YangZhengyi98/DreamRec/assets/82690004/191b3ed8-9ae4-4483-8cbe-240705e432b0)

![738b9706bb2f394526be4f7e76e2bfc](https://github.com/YangZhengyi98/DreamRec/assets/82690004/cfd80ecf-d0fd-46bb-9990-0ee521f065cd)

并且我在试验中也并未观测到ndcg@20高达0.008787的情况。

由于diffusion对参数极其敏感，因此在其他环境下可能调参范围和尺度会有所变化。

如果方便的话，可以提供一下你的GPU型号，我如果有相同型号的话可以调调看。

**Comment by Yewx0310** [comment]
This is an automatic reply, confirming that your e-mail was received. Thank you.--------------------------------------------Best WishesWenxue Ye.

**Comment by Yewx0310** [comment]
> 你好，
> 
> 可能是由于实验环境不同，在论文中给定的实验环境下(GPU type: 3090, python==3.7.13, pytorch==1.12.1, numpy==1.21.5)，按提供参数的实验结果为：
> 
> ![8660cd9c52f8d2dd1aaf4950c6131ae](https://private-user-images.githubusercontent.com/82690004/288843894-191b3ed8-9ae4-4483-8cbe-240705e432b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMjkxODMsIm5iZiI6MTcwMjAyODg4MywicGF0aCI6Ii84MjY5MDAwNC8yODg4NDM4OTQtMTkxYjNlZDgtOWFlNC00NDgzLThjYmUtMjQwNzA1ZTQzMmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA5NDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlN2ZjNzdiMzJlMDliOWUyYTY4MGU3M2RmOWMzZTJmZmFkNWNhYzA5ZjE4NjY2YjZlMGQ4MzcyYTBlNGM1M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UgtuXa-YLl_o2AUU8lwZ8A12YdABffFTTn7VCepoplM)
> 
> ![738b9706bb2f394526be4f7e76e2bfc](https://private-user-images.githubusercontent.com/82690004/288843917-cfd80ecf-d0fd-46bb-9990-0ee521f065cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMjkxODMsIm5iZiI6MTcwMjAyODg4MywicGF0aCI6Ii84MjY5MDAwNC8yODg4NDM5MTctY2ZkODBlY2YtZDBmZC00NmJiLTk5OTAtMGVlNTIxZjA2NWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA5NDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMWFjN2FkYTM2N2U2ZmZkNTg3OGNlNzZhZGUyNWE2ODdkNGIzZWI2ZDg1MTBmOWU4MGZiOTMxYWMzNTg5MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3qIxbX5cOhZQ20KmFHgC2cqHLGIEb24Qrk3UXxSqpEo)
> 
> 并且我在试验中也并未观测到ndcg@20高达0.008787的情况。
> 
> 由于diffusion对参数极其敏感，因此在其他环境下可能调参范围和尺度会有所变化。
> 
> 如果方便的话，可以提供一下你的GPU型号，我如果有相同型号的话可以调调看。

我的GPU是4090，但是我也有3090的GPU，我可以试试

**Comment by Hans2573** [comment]
我的GPU：Tesla V100
python 3.9.18
pytorch 2.1.1
numpy 1.24.4

**Comment by Smgdx2002** [comment]
作者你好，我想了解下对于yc和ks数据集，大概要训练多少个epoch才能差不多达到论文中的效果呢？
我大致训练了200个epoch，但是效果距离论文中提及的还是相差甚远。

**Comment by YangZhengyi98** [comment]
你好，我的实验中yc在100个epoch左右，ks应该不到50个epoch。

**Comment by GithubX-F** [comment]
I trained for 600 epochs, but never achieved a case where HR@20 was greater than 2.00% or NDCG@20 was greater than 0.70%, haha.🤡

**Comment by Dreamhepeng** [comment]
> 作者你好，我想了解下对于yc和ks数据集，大概要训练多少个epoch才能差不多达到论文中的效果呢？ 我大致训练了200个epoch，但是效果距离论文中提及的还是相差甚远。

你好，后面你是怎么查看ks的结果的呢？
